solution/planning_solutions.go

```diff
@@ -113,8 +114,15 @@ func FindBestPlanningSolutions(placeClusters [][]matching.Place, topSolutionsCou
 		top := heap.Pop(priorityQueue).(graph.Vertex)
 		res = append(res, top.Object.(PlanningSolution))
 	}
+	// min-heap, res needs to be reversed to get the descending order
+	return reverseArray(res)
+}
 
-	return res
+func reverseArray(arr []PlanningSolution) []PlanningSolution {
```

Comment:
Maybe put this function in utils and make it generic?

Reply:
Good suggestion, will do so.

Reply:
Moved this function to solution/utils.go, please check e9be6c7.
Wanted to use the more generic style with abstract input data type, but it seems require higher Go version (1.18+) to support.